src/app/question-seven/question-seven.component.scss

```diff
@@ -1,15 +1,49 @@
+a[href$=".com"] {
+    color: red;
+}
 #colorchilderen {
-    // overider this for second problem 
+    div:nth-child(1){
+        div:nth-child(1) div{
+            color: red;
+        }
+        div:nth-child(2) div{
+            color: yellow;
+        }
+    }
+    div{
+        color: green;
+    }
 }
 
-#bluecolor {
-    // for second problem
+#bluecolor  ul li:nth-child(2) {
+        color: blue;
 }
 
-#evenodd {
-    // for third problem 
+#evenodd li:nth-child(2n) {
+    color: red;
+}
+#evenodd li:nth-child(2n+1) {
+    color: green;
 }
 
-#lastselector {
-    // for fourth problem 
-}
\ No newline at end of file
+#lastselector ul li:last-child {
+    color: red;
+  }
+  #lastselector ul li ul li:last-child {
+    color: red;
+  }
+  #lastselector ul li ul li ul li:last-child {
+    color: red;
+  }
+  #lastselector ul li ul li ul li ul li:last-child {
```

Comment:
Why you did this? :slightly_frowning_face:

Reply:
The **ul** element in #lastselector has multiple **li** elements - 

- the **li** element at last will have color red 
   -  `#lastselector ul li:last-child`
- the first **li** element in **ul** has a nested **li**  i made last element of this **li** red using 
   - `#lastselector ul li ul li:last-child`
- similarly other li elements have nested ul elements and to make last li red i used 
    - `#lastselector ul li ul li ul li:last-child`  
    - `#lastselector ul li ul li ul li ul li:last-child`

Reply:
Does not `ul li:last-child` work?

Reply:
Yes, It does work I didn't think about it. you are right.

Reply:
Hi himanshu,
Thanks for showing interest, I am very sorry to inform you that giving position is closed.